src/JPEGView/PNGWrapper.cpp

```diff
@@ -62,12 +65,12 @@ struct FrameData {
 	unsigned int channels;
 	unsigned int frame_index;
 	png_uint_32 frame_count;
+	void* buffer;
+	size_t buffer_size;
 	int buffer_offset;
 };
 
-struct FrameData env = { 0 };
-size_t cached_buffer_size = 0;
-void* cached_buffer = NULL;
+PngReader::png_cache PngReader::cache = { 0 };
```

Comment:
Is this chunk of code supposed to live in the constructor?

Reply:
I think it makes more sense the way it is now. The contents of the cache don't need to be known ahead of time.